google-cloud-spanner/src/main/java/com/google/cloud/spanner/Mutation.java

```diff
@@ -358,6 +358,29 @@ public int hashCode() {
     return Objects.hash(operation, table, columns, values, keySet);
   }
 
+  private boolean areValuesEqual(List<Value> values, List<Value> otherValues) {
+    if (values == null && otherValues == null) {
+      return true;
+    } else if (values == null || otherValues == null) {
+      return false;
+    } else if (values.size() != otherValues.size()) {
+      return false;
+    } else {
+      for (int i = 0; i < values.size(); i++) {
+        final Value value = values.get(i);
+        final Value otherValue = otherValues.get(i);
+        if (!value.equals(otherValue) && (!isNaN(value) || !isNaN(otherValue))) {
```

Comment:
nit: maybe for future reference we should add a short comment on why we are doing this

Reply:
Good point, added a comment here